# iwinfo - Wireless Information Library - Command line frontend
#
#   Copyright (C) 2011 Jo-Philipp Wich <xm@subsignal.org>
#
# The iwinfo library is free software: you can redistribute it and/or
# modify it under the terms of the GNU General Public License version 2
# as published by the Free Software Foundation.
#
# The iwinfo library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
# See the GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with the iwinfo library. If not, see http://www.gnu.org/licenses/.

diff --git a/Makefile b/Makefile
index 715c03e..4e7943d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,19 +1,22 @@
 IWINFO_BACKENDS    = $(BACKENDS)
-IWINFO_CFLAGS      = $(CFLAGS) -std=gnu99 -fstrict-aliasing -Iinclude
-IWINFO_LDFLAGS     = -luci -lubox -lubus
+IWINFO_CFLAGS      = $(CFLAGS) -Wall -D_GNU_SOURCE -std=gnu99 -fstrict-aliasing -Iinclude -I/usr/local/include/libnl-tiny
+IWINFO_LDFLAGS     = -L$(TARGET_LIBS) -luci -lubox -lubus
 
 IWINFO_LIB         = libiwinfo.so
-IWINFO_LIB_LDFLAGS = $(LDFLAGS) -shared
+IWINFO_LIB_LDFLAGS = $(LDFLAGS) -Wl,-Bsymbolic-functions -shared
 IWINFO_LIB_OBJ     = iwinfo_utils.o iwinfo_wext.o iwinfo_wext_scan.o iwinfo_lib.o
 
-IWINFO_LUA         = iwinfo.so
-IWINFO_LUA_LDFLAGS = $(LDFLAGS) -shared -L. -liwinfo -llua
-IWINFO_LUA_OBJ     = iwinfo_lua.o
+IWINFO_LUA         = #iwinfo.so
+IWINFO_LUA_LDFLAGS = #$(LDFLAGS) -shared -L. -liwinfo -llua5.1
+IWINFO_LUA_OBJ     = #iwinfo_lua.o
 
 IWINFO_CLI         = iwinfo
 IWINFO_CLI_LDFLAGS = $(LDFLAGS) -L. -liwinfo
 IWINFO_CLI_OBJ     = iwinfo_cli.o
 
+ifneq ($(IWINFO_LUA),)
+	IWINFO_CFLAGS += -I/usr/include/lua5.1
+endif
 
 ifneq ($(filter wl,$(IWINFO_BACKENDS)),)
 	IWINFO_CFLAGS  += -DUSE_WL
@@ -36,10 +39,12 @@ endif
 %.o: %.c
 	$(CC) $(IWINFO_CFLAGS) $(FPIC) -c -o $@ $<
 
-compile: clean $(IWINFO_LIB_OBJ) $(IWINFO_LUA_OBJ) $(IWINFO_CLI_OBJ)
-	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_LIB_LDFLAGS) -o $(IWINFO_LIB) $(IWINFO_LIB_OBJ)
-	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_LUA_LDFLAGS) -o $(IWINFO_LUA) $(IWINFO_LUA_OBJ)
-	$(CC) $(IWINFO_LDFLAGS) $(IWINFO_CLI_LDFLAGS) -o $(IWINFO_CLI) $(IWINFO_CLI_OBJ)
+compile: clean $(IWINFO_LIB_OBJ) $(IWINFO_CLI_OBJ) $(IWINFO_LUA_OBJ)
+	$(CC) -o $(IWINFO_LIB) $(IWINFO_LIB_OBJ) $(IWINFO_LDFLAGS) $(IWINFO_LIB_LDFLAGS)
+	$(CC) -o $(IWINFO_CLI) $(IWINFO_CLI_OBJ) $(IWINFO_LDFLAGS) $(IWINFO_CLI_LDFLAGS)
+ifneq ($(WINFO_LUA),)
+	$(CC) -o $(IWINFO_LUA) $(IWINFO_LUA_OBJ) $(IWINFO_LDFLAGS) $(IWINFO_LUA_LDFLAGS)
+endif
 
 clean:
-	rm -f *.o $(IWINFO_LIB) $(IWINFO_LUA) $(IWINFO_CLI)
+	rm -f *.o $(IWINFO_LIB) $(IWINFO_CLI) $(IWINFO_LUA)
